File: examples/deno/json_parser_test.ts
import { assertEquals } from "jsr:@std/assert@^1.0.0";
import { parseConfig } from "./json_parser.ts";

const configUrl = new URL("./deno_project_matrix.json", import.meta.url);

Deno.test("parses the JSON file", () => {
  const result = parseConfig(configUrl);
  assertEquals(typeof result, "object");
  assertEquals(result !== null, true);
});

Deno.test("parses the expected values", () => {
  const result = parseConfig(configUrl) as Record<string, unknown>;

  const os = result["os"] as string[];
  assertEquals(os.includes("ubuntu-latest"), true);
  assertEquals(os.includes("windows-latest"), true);
  assertEquals(os.includes("macos-latest"), true);

  const versions = result["versions"] as Record<string, string[]>;
  assertEquals(versions["deno"], ["v1.x", "v2.x"]);

  assertEquals(result["ghpages_branch"], "ghgapes");
});

Deno.test("returns null for a missing file", () => {
  const result = parseConfig("non-existent.json", true);
  assertEquals(result, null);
});

File: examples/java/src/test/java/io/github/json2varssetter/JsonParserTest.java
package io.github.json2varssetter;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import com.fasterxml.jackson.databind.JsonNode;
import java.nio.file.Paths;
import org.junit.jupiter.api.Test;

class JsonParserTest {

    private final String configPath =
        Paths.get("java_project_matrix.json").toString();

    @Test
    void parsesJsonFile() {
        JsonNode result = JsonParser.parseConfig(configPath);
        assertNotNull(result);
    }

    @Test
    void parsesExpectedValues() {
        JsonNode result = JsonParser.parseConfig(configPath);
        assertNotNull(result);

        JsonNode os = result.get("os");
        assertTrue(os.toString().contains("ubuntu-latest"));
        assertTrue(os.toString().contains("windows-latest"));
        assertTrue(os.toString().contains("macos-latest"));

        JsonNode versions = result.get("versions").get("java");
        assertEquals("[\"11\",\"17\",\"21\"]", versions.toString());

        assertEquals("ghgapes", result.get("ghpages_branch").asText());
    }

    @Test
    void returnsNullForMissingFile() {
        JsonNode result = JsonParser.parseConfig("non-existent.json", true);
        assertNull(result);
    }
}
